### ISSUE TITLE
Why am i getting cors error?

### ISSUE THREAD

**Issue by shizzic** [gin-gonic/gin#3001]
- With issues:
  - Use the search tool before opening a new issue.
  - Please provide source code and commit sha if you found a bug.
  - Review existing issues and provide feedback or react to them.

## Description

I use git-contrib cors package to specify allow-origins, but whatever i did, nothing changed. And I using Default() method to do this (just for test).
It seems like this settings just don't aplly or something, 'cause i can use postman and he works fine, but not my client side on js.
Just see that middleware not working too, just adding headers manually in EACH handle request do what i need. 

## How to reproduce

```
package main

import (
	"github.com/gin-contrib/cors"
	"github.com/gin-gonic/gin"
)

// Run both: http and https servers
func run() {
	r.Use(cors.Default())
	http.Use(cors.Default())
	http.Use(redirect()) // bind endless redirect for NONE https requests
	go r.RunTLS("ip:443", "path", "path")
	http.Run("ip:80")
}

```

## Expectations

```
Just wanna do simple fetch for test, that return some string, that's all.
```

## Actual result

```
Access to fetch at 'https://my-domai.some' from origin 'http://localhost:8080' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

## Environment

- go version: 1.17.3
- gin version (or commit ref): 1.7.7
- operating system: Ubuntu 20.04

**Comment by shizzic** [comment]
So, i found an answer:
[Just check lust answer](https://stackoverflow.com/questions/29418478/go-gin-framework-cors)
It's detail that missing litarelly EACH guide.
